components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/federatedIdpInitLogout/factory/LogoutRequestFactoryTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc.federatedIdpInitLogout.factory;
+
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.CarbonBaseConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.nio.file.Paths;
+import java.util.Collections;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.testng.Assert.assertEquals;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * Unit test class for  LogoutRequestFactory
+ */
+@PrepareForTest({PrivilegedCarbonContext.class})
+@PowerMockIgnore("jdk.internal.reflect.*")
+public class LogoutRequestFactoryTest extends PowerMockTestCase {
+
+    @Mock
+    HttpServletRequest mockHttpServletRequest;
+
+    @Mock
+    HttpServletResponse mockHttpServletResponse;
+    
+    LogoutRequestFactory logoutRequestFactory;
+
+    private static final String CARBON_HOME =
+            Paths.get(System.getProperty("user.dir"), "src", "test", "resources").toString();
+
+    @BeforeTest
+    public void init() {
+
+        System.setProperty(CarbonBaseConstants.CARBON_HOME, CARBON_HOME);
+        logoutRequestFactory = new LogoutRequestFactory();
+    }
+
+    @DataProvider(name = "canHandleDataProvider")
+    public Object[][] getHttpServletRequestURI() {
+
+        return new String[][]{{"http://localhost:9443/identity/oidc/slo", "true"}, {"http://localhost:9443/identity" +
```

Comment:
Can we add a test case for the tenant scenario?

ie, a URL like http://localhost:9443/t/hoho.com/identity/oidc/slo

Also do we allow patterns like http://localhost:9443/identity/oidc/slo?somequery=value

Reply:
Added a test case. 
OIDC specification states the following/

> The POST body MAY contain other values in addition to logout_token. Values that are not understood by the implementation MUST be ignored.